tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -4523,40 +4525,61 @@ Status ConvertPack(OpConverterParams* params) {
         node_def.name());
   }
 
-  // Validate inputs. Values must be tensors for now.
-  std::vector<std::pair<string, bool>> inputs_is_weight;
+  TrtInputArg expected_arg =
+      params->use_implicit_batch ? TrtInputArg::kTensor : TrtInputArg::kBoth;
+
+  std::vector<std::pair<string, TrtInputArg>> inputs_is_weight;
   for (int i = 0; i < num_inputs; ++i) {
-    inputs_is_weight.push_back({StrCat("values_", i), false});
+    inputs_is_weight.push_back({StrCat("values_", i), expected_arg});
   }
   TF_RETURN_IF_ERROR(CheckInputsWeights(*params, inputs_is_weight));
 
-  // TODO(hinsu): Enable INT32 with TensorRT version 5.1.3 after testing.
-  TF_RETURN_IF_ERROR(
-      AllowDataTypes(*params, {DataType::DT_FLOAT, DataType::DT_HALF}));
-
+#if IS_TRT_VERSION_GE(6, 0, 0, 0)
+  std::set<DataType> allowed_types{DataType::DT_FLOAT, DataType::DT_HALF,
+                                   DataType::DT_INT32};
+#else
+  std::set<DataType> allowed_types{DataType::DT_FLOAT, DataType::DT_HALF};
+#endif
+  TF_RETURN_IF_ERROR(AllowDataTypes(*params, allowed_types));
   if (num_inputs > 1) {
     // Verify that inputs are compatible for concatenation after the expansion.
     TF_RETURN_IF_ERROR(
         VerifyShapesMatch(inputs, /*masked_dim=*/-1, node_def.name()));
   }
 
-  // Convert axis from the TensorFlow format to TensorRT format.
-  const nvinfer1::Dims dims = inputs.at(0).GetTrtDims();
+  // Convert axis from the TensorFlow format to TensorRT format. First we check
+  // if any of the inputs is a weight. If yes then we use its shape, because
+  // it is conversion time constant, otherwise we just select the first input.
+  int idx = 0;
+  for (int i = 0; i < inputs.size(); i++) {
+    if (inputs.at(i).is_weights()) {
```

Comment:
Can you change this to find an input with all dimension values known instead? It is possible that there are multiple non-weights inputs, some with dynamic dimension values but others with static dimension values, right?

Can you also add a test case for selecting a tensor with static shape?

Reply:
That is a good point. Yes, it is possible to have multiple non-weight inputs where some could have static shape. I have changed the condition and added a test (the last test).